### PR TITLE
#178  Master / Intern / Other role permissions

### PR DESCRIPTION
**Pull request by vryazanov**
#178 
* checking permissions for Master / Intern / Other roles
* add tests
* fix docker setup

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#347](https://github.com/timelesslounge/timelessis/issues/347) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#347](https://github.com/timelesslounge/timelessis/issues/347) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/abbf4d37-6160-4913-9320-814fcbe5cb2d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/conftest.py [hunk 1]
```diff
@@ -14,14 +14,19 @@ def app():
...
-    DB.create_all()
     yield app
-    DB.session.remove()
-    DB.drop_all()
     os.close(db_fd)
     os.unlink(db_path)
 
 
+@pytest.fixture(autouse=True)
+def flush_db(app):
```
Comment (vladarefiev) [inline tests/conftest.py]: @vryazanov God bless you

File: tests/integration/it_other_privilages_test.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    Method.CREATE,
+    Method.UPDATE,
+    Method.DELETE
+))
+def test_can_access_to_self_account(method, app):
+    employee = factories.EmployeeFactory()
+    flask.g.user = employee
+    assert has_privilege(
+        method=method, resource="employee", employee_id=employee.id)
+
```
Comment (vladarefiev) [inline tests/integration/it_other_privilages_test.py]: @vryazanov let's add here
```
assert not has_privilege(
        method=method, resource="employee", employee_id=-1)
```
and here will be no copy/paste code. WDYT?

**Comment by vryazanov** [comment]
@vladarefiev fixed, pls have a look

**Comment by vladarefiev** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 14 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#347](https://github.com/timelesslounge/timelessis/issues/347) is not in the agenda of @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), can't inspect

**Comment by 0crat** [comment]
The job [#347](https://github.com/timelesslounge/timelessis/issues/347) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/347#issuecomment-466183837))

@ypshenychka This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#347](https://github.com/timelesslounge/timelessis/issues/347) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 11hr
  * @vladarefiev[/z](https://www.0crat.com/u/vladarefiev) will receive 20 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23347%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@vladarefiev  According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vladarefiev** [comment]
@ypshenychka Confirm

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**